src/helpers/MdObjectHelper.ts

```diff
@@ -83,8 +83,14 @@ export const mdObjectToPivotBucketProps = (
 
     const filters: VisualizationInput.IFilter[] = (afm.filters || []).filter(afmFilter => {
         // Filter out expression filters which are not supported in bucket interface
-        return AFM.isDateFilter(afmFilter) || AFM.isAttributeFilter(afmFilter);
-    }) as AFM.FilterItem[];
+        return (
+            AFM.isDateFilter(afmFilter) ||
+            AFM.isAttributeFilter(afmFilter) ||
+            VisualizationObject.isMeasureValueFilter(
```

Comment:
This seems strange that we test the others by `AFM.` functions but this one by `VisualizationObject .` Anyway, shouldn't we exclude just the expression filter as described in the comment instead of including all the other filters?

Reply:
Whole concept of what is happening here is strange. It works with Visualization object methods and uses the fact that apart from some exceptions (Measure value filter added now) it was the same shape as AFM filters. The reason for this is to reuse the whole set of functions that merge filters into AFM :/ And this whole construction is here probably because someone was lazy to create `isExpressionFilter` type guard that does not exist. I will create it and use it instead.

Reply:
I updated the typings with missing type guards and rewrote this piece of code. You were right, it was a mess and it makes way more sense now.

However, I changed the opt-in vs opt-out logic that could create some issue in the future if a new incompatible type of test would be introduced. It should be fine imho.